src/pubsub/ua_pubsub_ns0.c

```diff
@@ -1849,6 +1849,149 @@ setSecurityKeysAction(UA_Server *server, const UA_NodeId *sessionId, void *sessi
 
 #endif
 
+#if defined(UA_ENABLE_PUBSUB_SKS) && defined(UA_ENABLE_PUBSUB_INFORMATIONMODEL_METHODS)
+static UA_StatusCode
+getSecurityKeysAction(UA_Server *server, const UA_NodeId *sessionId, void *sessionHandle,
+                      const UA_NodeId *methodId, void *methodContext,
+                      const UA_NodeId *objectId, void *objectContext, size_t inputSize,
+                      const UA_Variant *input, size_t outputSize, UA_Variant *output) {
+    /*Check whether the channel is encrypted according to specification*/
+    session_list_entry *session_entry;
+    LIST_FOREACH(session_entry, &server->sessions, pointers) {
+        if(UA_NodeId_equal(&session_entry->session.sessionId, sessionId)) {
+            if(session_entry->session.header.channel->securityMode !=
+               UA_MESSAGESECURITYMODE_SIGNANDENCRYPT)
+                return UA_STATUSCODE_BADSECURITYMODEINSUFFICIENT;
+        }
+    }
+
+    if(!server || !input)
+        return UA_STATUSCODE_BADINVALIDARGUMENT;
+
+    if(inputSize < 3)
```

Comment:
Check the outputsize as well.
This is set in the information model.
A badly configured AccessControl policy might allow a user to modify the output size.
Just to make sure...